### PR TITLE
Loki: Rename "explain" toggle to "explain query"

### LINKED ISSUE #59222: Loki Query Editor: rename "Explain" toggle to "Explain query".
# Context

In the Query Editor, we have a toggle to explain the query, which currently says "Explain", which is ambiguous and potentially unclear.

imagen

# Expectation

- Rename it to `Explain query`.

### PR DESCRIPTION
**Pull request by itsgareth**
**What is this feature?**
we are renaming the toggle to explain the query from `"explain"` to `"explain query"`

**Why do we need this feature?**
currently it can be a little confusing, what exactly will be explained if i toggle this on.
with this change it removes any ambiguity and clearly states what will be explained.

**Who is this feature for?**
...

**Which issue(s) does this PR fix?**:
Fixes #59222

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by matyax** [comment]
I agree with this change, but I'd like to know that the rest of the squad is on board with this change.

**Comment by svennergr** [comment]
> I agree with this change, but I'd like to know that the rest of the squad is on board with this change.

I am fine with the change, but we might want to ask @grafana/observability-metrics if we should change this in Prom as well. Also I am not sure if we need to change the documentation (https://github.com/grafana/grafana/blob/main/docs/sources/datasources/loki/query-editor/index.md#explain-mode) to reflect the change.

**Comment by gwdawson** [comment]
> > I agree with this change, but I'd like to know that the rest of the squad is on board with this change.
> 
> I am fine with the change, but we might want to ask @grafana/observability-metrics if we should change this in Prom as well to be consistent in both query builders. Also I am not sure if we need to change the documentation (https://github.com/grafana/grafana/blob/main/docs/sources/datasources/loki/query-editor/index.md#explain-mode) to reflect the change.
> 
> Based on the above documentation we also could change the text to `explain mode`.

@svennergr thanks for bring the docs to attention, just realised i also need to edit the docs for the `Raw query` too, since now that is always enabled and there is no toggle.

as for whether to use `explain query` and change the docs, or use `explain mode` and leave the docs, i will ask in the logs channel and get the opinion from the rest of the squad.

File: docs/sources/datasources/loki/query-editor/index.md [hunk 1]
```diff
@@ -125,9 +125,9 @@ To re-order operations manually, drag the operation box by its name and drop it
 
 In same cases the query editor can detect which operations would be most appropriate for a selected log stream. In such cases it will show a hint next to the `+ Operations` button. Click on the hint to add the operations to your query.
 
-### Explain mode
+### Explain Query
```
Comment (ivanahuckova) [inline docs/sources/datasources/loki/query-editor/index.md:128]:
```suggestion
### Explain query
```

We should use sentence case capitalisation based on style guide: https://grafana.com/docs/writers-toolkit/style-guide/capitalization-punctuation/#sentence-case-capitalization

File: docs/sources/datasources/loki/query-editor/index.md [hunk 2]
```diff
@@ -125,9 +125,9 @@ To re-order operations manually, drag the operation box by its name and drop it
...
 
-Explain mode helps with understanding the query. It shows a step by step explanation of all query parts and the operations.
+This section is only shown if the `Explain Query` switch from the query editor top toolbar is set to `on`. It shows a step by step explanation of all query parts and the operations.
```
Comment (ivanahuckova) [inline docs/sources/datasources/loki/query-editor/index.md:130]:
```suggestion
This section is only shown if the `Explain query` switch from the query editor top toolbar is set to `on`. It shows a step by step explanation of all query parts and the operations.
```

File: public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx [hunk 3]
```diff
@@ -136,7 +136,7 @@ export const LokiQueryEditor = React.memo<LokiQueryEditorProps>((props) => {
             Label browser
           </Button>
         </Stack>
-        <QueryHeaderSwitch label="Explain" value={explain} onChange={onExplainChange} />
+        <QueryHeaderSwitch label="Explain Query" value={explain} onChange={onExplainChange} />
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx:139]:
```suggestion
        <QueryHeaderSwitch label="Explain query" value={explain} onChange={onExplainChange} />
```

**Review by ivanahuckova — APPROVED** [review]
LGTM! 👍

**Review by ivanahuckova — APPROVED** [review]
1 more idea, would you check with @grafana/observability-metrics if it would be okay with them to change the name of toggle also for Prometheus, and if they say yes, you can either do it in this PR, or next PR.

**Comment by itsmylife** [comment]
> 1 more idea, would you check with @grafana/observability-metrics if it would be okay with them to change the name of toggle also for Prometheus, and if they say yes, you can either do it in this PR, or next PR.

We are ok :)

**Pull request merged**